build-tools/xaprepare/xaprepare/Steps/Step_ThirdPartyNotices.cs

```diff
@@ -0,0 +1,201 @@
+
+THIRD - PARTY SOFTWARE NOTICES AND INFORMATION
+Do Not Translate or Localize
+
+Xamarin-Android incorporates components from the projects listed below.
+Microsoft licenses these components to you under Microsoft’s licensing
+terms, except that components licensed under open source licenses
+requiring that such components remain under their original license are
+being made available to you under their original licensing terms.
+The original copyright notices and the licenses under which Microsoft
+received such components are set forth below for informational purposes.
+Microsoft reserves all rights not expressly granted herein, whether by
+implication, estoppel or otherwise."			}
+		};
+
+		public Step_ThirdPartyNotices ()
+			: base ("Preparing Third Party Notices")
+		{}
+
+#pragma warning disable CS1998
+		protected override async Task<bool> Execute (Context context)
+		{
+			GenerateThirdPartyNotices (Path.Combine (BuildPaths.XamarinAndroidSourceRoot, "ThirdPartyNotices.txt"),
+			                           ThirdPartyLicenseType.Foundation,
+			                           includeExternalDeps: false,
+			                           includeBuildDeps: false);
+			Log.StatusLine ();
+			GenerateThirdPartyNotices (Path.Combine (context.XAInstallPrefix, "ThirdPartyNotices.txt"),
+			                           ThirdPartyLicenseType.MicrosoftOSS,
+			                           includeExternalDeps: true,
+			                           includeBuildDeps: false);
+
+			return true;
+		}
+#pragma warning restore CS1998
+
+		void GenerateThirdPartyNotices (string outputPath, ThirdPartyLicenseType licenseType, bool includeExternalDeps, bool includeBuildDeps)
+		{
+			List<Type> types = Utilities.GetTypesWithCustomAttribute<TPNAttribute> ();
+			if (types.Count == 0) {
+				Log.StatusLine ("No Third Party Notice entries found", ConsoleColor.Gray);
+				return;
+			}
+
+			var licenses = new SortedDictionary <string, ThirdPartyNotice> (StringComparer.OrdinalIgnoreCase);
+			foreach (Type type in types) {
+				EnsureValidTPNType (type);
+
+				if (type.IsSubclassOf (typeof (ThirdPartyNoticeGroup))) {
+					ProcessTPN (licenses, Activator.CreateInstance (type) as ThirdPartyNoticeGroup, includeExternalDeps, includeBuildDeps);
+					continue;
+				}
+
+				if (type.IsSubclassOf (typeof (ThirdPartyNotice))) {
+					ProcessTPN (licenses, Activator.CreateInstance (type) as ThirdPartyNotice, includeExternalDeps, includeBuildDeps);
+					continue;
+				}
+
+				throw new NotSupportedException ($"ThirdPartyNotice type {type.FullName} not supported");
+			}
+
+			if (licenses.Count == 0)
+				return;
+
+			string blurb;
+			if (!tpnBlurbs.TryGetValue (licenseType, out blurb))
+				throw new InvalidOperationException ($"Unknown license type {licenseType}");
+
+			using (StreamWriter sw = Utilities.OpenStreamWriter (outputPath)) {
+				Log.StatusLine ($" Generating: {Utilities.GetRelativePath (BuildPaths.XamarinAndroidSourceRoot, outputPath)}", ConsoleColor.Gray);
```

Comment:
@grendello Given the recent Azure Pipeline [macOS build failure](https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=2732325&view=logs&s=1afc3bfe-122c-538b-e9ad-2a86c2efcfef&j=9fac13c5-5013-5cc9-5c17-da47b3873a81) perhaps we should just print the full `outputPath` here in case we're somehow putting these files in the wrong place?

Reply:
I'll log the full path in using the `Debug` severity, it will show up in the logs.